ecleankernel/__main__.py

```diff
@@ -256,6 +261,34 @@ def main(argv: typing.List[str]) -> int:
                 bootloader=bootloader,
                 destructive=args.destructive)
 
+            if HOOK_DIR.is_dir():
+                preserved_kernels = [k for k in kernels
+                                     if k not in removals]
+
+                def as_version_list(kernels: List[Kernel]) -> str:
+                    return " ".join(list(map(lambda x: x.version, kernels)))
+
+                env = dict()
+
+                preserved_str = as_version_list(preserved_kernels)
+                env["ECLEAN_KERNEL_PRESERVED"] = preserved_str
+
+                removal_str = as_version_list(list(removals.keys()))
```

Comment:
Converting it into a list makes no sense here, just accept an iterable in the function.